packages/extension-ui/src/Popup/Derive.tsx

```diff
@@ -16,22 +16,42 @@ export default function Derive ({ match: { params: { address: parentAddress } }
   const [account, setAccount] = useState<null | { address: string; suri: string }>(null);
   const [name, setName] = useState<string | null>(null);
   const [password, setPassword] = useState<string | null>(null);
+  const [parentPassword, setParentPassword] = useState<string | null>(null);
+  const [isProperParentPassword, setIsProperParentPassword] = useState(false);
 
   const _onCreate = useCallback(async () => {
-    if (!account || !name || !password) {
+    if (!account || !name || !password || !parentPassword) {
       return;
     }
-    await deriveAccount(parentAddress, account.suri, name, password);
+    await deriveAccount(parentAddress, account.suri, parentPassword, name, password);
     onAction('/');
   }, [account, name, password]);
 
+  const _onParentPasswordEnter = (enteredPassword: string): void => {
+    setParentPassword(enteredPassword);
+    exportAccount(parentAddress, enteredPassword)
```

Comment:
I'm not fully convinced of using `exportAccount` to validate the password.

Pros:
 - simple and works :-)
Cons:
 - if the `Extension.accountsExport` function changes (eg. won't throw exception on wrong password), this won't work
 - a bit unclear for the reader here

Reply:
Also: wrap this into useCallback